File: python/Euler9.py

def euler9():
  possible_triples = []
  for i in range(1000):
    for j in range(i, 1000):
      for k in range(j+1, 1000):
        if i + j + k == 1000:
          possible_triples.append((i,j,k))

  for a,b,c in possible_triples:
    if a**2 + b**2 == c**2:
      return a*b*c

File: Haskell/Euler9.hs
import Data.List

-- The python implementation is much quicker.
pythagoreanTriple1000 :: Maybe Integer
pythagoreanTriple1000 = fmap product_tuple . find adds_to_1000 $ possible_triples
  where
    product_tuple (a,b,c) = a * b * c
    adds_to_1000 (a, b, c) = (a^2 + b^2 == c^2) 
    possible_triples = [(x,y,z) | x <- [1..1000], y <- [1..1000], z <- [1..1000], x + y + z == 1000, x <= y, y <= z]
